### PR TITLE
Warnings elimination

### PR DESCRIPTION
**Pull request by AnastaZIuk**
status:

- entry warnings amount: **over 6073**
- flags: **-Wall**
- compiler: **msvc**

### CODE REVIEW

File: include/irr/asset/IAssetManager.h [hunk 1]
```diff
@@ -599,7 +599,7 @@ class IAssetManager : public core::IReferenceCounted
             const char** exts = _writer->getAssociatedFileExtensions();
             for (uint32_t i = 0u; i < IAsset::ET_STANDARD_TYPES_COUNT; ++i)
             {
-                const IAsset::E_TYPE type = IAsset::E_TYPE(1u << i);
+                const IAsset::E_TYPE type = IAsset::E_TYPE((uint64_t)1 << (uint64_t)i);
```
Comment (devshgraphicsprogramming) [inline include/irr/asset/IAssetManager.h:602]: actually if you want a 64bit uint constant the suffix is `ull`, e.g. `1ull`

File: include/irr/core/math/plane3dSIMD.h [hunk 2]
```diff
@@ -98,7 +98,7 @@ class plane3dSIMDf : private vectorSIMDf
 		}
 
 		//!
-        static inline plane3dSIMDf transform(const plane3dSIMDf& _in, const matrix3x4SIMD& _mat)
+        static inline [[maybe_unused]] plane3dSIMDf transform(const plane3dSIMDf& _in, const matrix3x4SIMD& _mat)
```
Comment (devshgraphicsprogramming) [inline include/irr/core/math/plane3dSIMD.h:101]: i dont think that function declaration return type is the place to put `[[maybe_unused]]`

File: include/irr/asset/IAssetManager.h (hunk 1 above)
Comment (AnastaZIuk) [reply include/irr/asset/IAssetManager.h:602]: I know those literals, right

File: include/irr/core/math/plane3dSIMD.h (hunk 2 above)
Comment (AnastaZIuk) [reply include/irr/core/math/plane3dSIMD.h:101]: I didn't solve the problem though, so it will be removed probably

File: include/quaternion.h [hunk 3]
```diff
@@ -382,9 +382,9 @@ inline void quaternion::toEuler(vector3df_SIMD& euler) const
...
 	{
 		// heading = rotation about z-axis
-		euler.Z = (float) atan2(2.0 * (X*Y +Z*W),(sqr.X - sqr.Y - sqr.Z + sqr.W));
+		euler.Z = (float) atan2(2.0f * (X*Y + Z*W),(sqr.X - sqr.Y - sqr.Z + sqr.W));
 		// bank = rotation about x-axis
-		euler.X = (float) atan2(2.0 * (Y*Z +X*W),(-sqr.X - sqr.Y + sqr.Z + sqr.W));
+		euler.X = (float) atan2(2.0f * (Y*Z +X*W),(-sqr.X - sqr.Y + sqr.Z + sqr.W));
 		// attitude = rotation about y-axis
 		euler.Y = (float) asin( core::clamp(test, -1.0, 1.0) );
 	}
```
Comment (devshgraphicsprogramming) [inline include/quaternion.h:390]: actually while you're at it, this function should be changed to use floats throughout

File: include/irr/core/math/plane3dSIMD.h (hunk 2 above)
Comment (AnastaZIuk) [reply include/irr/core/math/plane3dSIMD.h:101]: but probably should be at def

**Comment by devshgraphicsprogramming** [comment]
Prefer `static_cast<T>(var)` to C-style cast `(T)var`, C-style cast will cast anything to anything pretty much !

careful with `decltype(variable)` especially with regards to ints, and similar, because `decltype(var)` on something that was declared as `T var` actually evaluates to `T&` or `const T&` (depending on current scope's access const qualifiers), and you really dont want to be casting constant literals to references (this ties into why you want to use `static_cast` and not C-style cast).

You'd actually need a std::decay for that... ask @Crisspl how to best handle

File: include/irr/core/math/plane3dSIMD.h (hunk 2 above)
Comment (devshgraphicsprogramming) [reply include/irr/core/math/plane3dSIMD.h:101]: its probably being unused where its being called.

File: include/irr/asset/IAssetManager.h (hunk 1 above)
Comment (devshgraphicsprogramming) [reply include/irr/asset/IAssetManager.h:602]: so why the cast?

File: include/irr/core/math/plane3dSIMD.h (hunk 2 above)
Comment (AnastaZIuk) [reply include/irr/core/math/plane3dSIMD.h:101]: yes, I wanted to use attribute to avoid it

File: include/irr/asset/IAssetManager.h (hunk 1 above)
Comment (AnastaZIuk) [reply include/irr/asset/IAssetManager.h:602]: to fix

**Comment by AnastaZIuk** [comment]
> Prefer `static_cast<T>(var)` to C-style cast `(T)var`, C-style cast will cast anything to anything pretty much !
> 
> careful with `decltype(variable)` especially with regards to ints, and similar, because `decltype(var)` on something that was declared as `T var` actually evaluates to `T&` or `const T&` (depending on current scope's access const qualifiers), and you really dont want to be casting constant literals to references (this ties into why you want to use `static_cast` and not C-style cast).
> 
> You'd actually need a std::decay for that... ask @Crisspl how to best handle

I need to change for `static_cast`, but I find it probably not true (quite) with evaluating to `T&` or `const T&`, since it depends on the way you call decltype

**Comment by devshgraphicsprogramming** [comment]
> I need to change for static_cast, but I find it probably not true (quite) with evaluating to T& or const T&, since it depends on the way you call decltype

I think `static_cast<decltype()>` should protect us against most things

**Comment by AnastaZIuk** [comment]
> > I need to change for static_cast, but I find it probably not true (quite) with evaluating to T& or const T&, since it depends on the way you call decltype
> 
> I think `static_cast<decltype()>` should protect us against most things

yeah, it's true

File: include/irr/core/math/plane3dSIMD.h (hunk 2 above)
Comment (devshgraphicsprogramming) [reply include/irr/core/math/plane3dSIMD.h:101]: its the correct place to put it, where its being called and unused, not the global definition for everyone

**Comment by devshgraphicsprogramming** [comment]
Update to match the HEAD of `shader_pipeline` fix my comments and I'll merge

**Comment by devshgraphicsprogramming** [comment]
closing temporarily.... reopen when ready for merge

**Pull request closed without merging**